### DISCUSSION TITLE
!include <dir> causes a DSL parser exception

### DISCUSSION THREAD

**Discussion by okamiraner** [structurizr/cli#136] [Q&A]
### Question

Hello

Documentation says that I can use `!include` for including all .dsl files inside the directory. In fact, when trying to export a workspace, I get the following exception randomly:

`$ structurizr-cli export -workspace workspace.dsl -format mermaid`

```
...
com.structurizr.dsl.StructurizrDslParserException: Input length = 1 at line 3: !include model
	at com.structurizr.dsl.StructurizrDslParser.parse(StructurizrDslParser.java:901)
	at com.structurizr.dsl.StructurizrDslParser.parse(StructurizrDslParser.java:128)
	at com.structurizr.cli.export.ExportCommand.run(ExportCommand.java:131)
	at com.structurizr.cli.StructurizrCliApplication.run(StructurizrCliApplication.java:77)
	at com.structurizr.cli.StructurizrCliApplication.main(StructurizrCliApplication.java:112)
...
```

Besides, including a single file works 100% of the time, but in breaks he whole point of a modular workspace adding excessive references that duplicate the folder content.

Its is hard to identify the problem, since the behaviour changes in different runs with the same files.

My directory tree:

```
├── model
│   ├── person.dsl
│   └── workspace.dsl
├── view
│   └── workspace.dsl
└── workspace.dsl
```

and the root file:

```
workspace {
	model {
		!include model
	}

	views {
		!include view
	}
}
```

**Comment by simonbrowndotje** [comment]
> Its is hard to identify the problem, since the behaviour changes in different runs with the same files.

Feel free to include a complete example to reproduce this (i.e. all of the DSL files, perhaps in a ZIP file), otherwise there's not much help I can offer I'm afraid. It would also be good to know which version of the CLI you're using.

**Reply by okamiraner** [reply]
Sure, here the [archive](https://github.com/structurizr/cli/files/12144612/workspace.zip) and my version is `structurizr-cli: 1.30.0`.

**Comment by mawid6** [comment]
I am having the same issue. `!include` works for a specific file, regardless of if the file is in the same or a child directory.

However, I cannot get `!include` to work on a directory.

### So, this works:
```
!include sources/kp-users-test.dsl
```
### But this does not:
```
!include sources
```

I have tried different shells, different versions of the include statemnt, e g `/sources`, slash at end, etc... To no avail.

Any tips?

#### Stacktrace 
The stacktrace looks to be the same as @okamiraner got above, line numbers are a tiny bit different likely due updated version...

```
C:\code\cbf-re-ipex-kp-planner>structurizr export -f plantuml -w kp-overview-c4ws.dsl -o c4model
Exporting workspace from kp-overview-c4ws.dsl
 - loading workspace from DSL
com.structurizr.dsl.StructurizrDslParserException: Input length = 1 at line 13 of C:\code\cbf-re-ipex-kp-planner\kp-overview-c4ws.dsl: !include sources
        at com.structurizr.dsl.StructurizrDslParser.parse(StructurizrDslParser.java:912)
        at com.structurizr.dsl.StructurizrDslParser.parse(StructurizrDslParser.java:128)
        at com.structurizr.cli.export.ExportCommand.run(ExportCommand.java:131)
        at com.structurizr.cli.StructurizrCliApplication.run(StructurizrCliApplication.java:81)
        at com.structurizr.cli.StructurizrCliApplication.main(StructurizrCliApplication.java:119)
```

**Reply by simonbrowndotje** [reply]
Feel free to open a new issue with the relevant details (e.g. CLI version number, example workspace to reproduce the issue, etc).
